twine/utils.py

```diff
@@ -50,21 +50,53 @@
 logger = logging.getLogger(__name__)
 
 
+def _parse_file(path: str, **open_kwargs: Any) -> configparser.RawConfigParser:
+    """Open and parse a configuration file.
+
+    This helper performs a single open/read operation so that if a
+    UnicodeDecodeError is raised it happens before the parser has been
+    partially populated.
+    """
+    parser = configparser.RawConfigParser()
+    with open(path, **open_kwargs) as f:
+        parser.read_file(f)
+    return parser
+
+
+def _parse_config(path: str) -> configparser.RawConfigParser:
+    """Parse a config file with a UTF-8 fallback on decode errors.
+
+    Try to parse using the default system encoding first; if a
+    UnicodeDecodeError occurs, retry using UTF-8 and log that a fallback
+    was used.
+    """
+    logger.info("Parsing configuration from %s", path)
+    try:
+        parser = _parse_file(path)
+    except UnicodeDecodeError:
+        logger.info("Configuration file not readable with default locale encoding, trying UTF-8")
+    else:
+        return parser
```

Comment:
```suggestion
        return parser

```

Personally, I find an extra new-line here makes it easier to parse the fact that there's different logical blocks. In other words, there's one `try/except/else` and a second `try/except/else`.